### PR TITLE
Common templates and move README content

### LINKED ISSUE #1245: Clarify google api java client documentation
It's not an actual bug but seems like the documentation is a bit outdated. 
The page I am talking about is https://developers.google.com/api-client-library/java/google-api-java-client/oauth2 .

 It was updated on 09.13.2018 . However it still mentions google plus related stuff which I think it's not relevant anymore or at least it's very confusing as Google+ APIs are shutting down.

Could you please verify if that's the case?

### LINKED ISSUE #1264: where are (or how to build) the generated libraries?
The official web page https://developers.google.com/api-client-library/java/ still references the old version 1.25.0.

The latest official generated library for Drive API
```
  <dependency>
      <groupId>com.google.apis</groupId>
      <artifactId>google-api-services-drive</artifactId>
      <version>v2-rev361-1.25.0</version>
    </dependency>
```

But, I am using version 1.28.0 I found in maven repo for some libs:
```
  <dependency>
      <groupId>com.google.apis</groupId>
      <artifactId>google-api-services-drive</artifactId>
      <version>v2-rev20190501-1.28.0</version>
    </dependency>
```
What is the correct and official version of the generated libraries? 

Or should we generate our own?

### LINKED ISSUE #1341: Publish SNAPSHOT releases nightly
This will help us detect downstream breakage before releasing the library

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #1341 
Fixes #1264 
Fixes #1245

### CODE REVIEW

File: .kokoro/continuous/dependencies.cfg [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Format: //devtools/kokoro/config/proto/build.proto
+
+# Configure the docker image for kokoro-trampoline.
+env_vars: {
+    key: "TRAMPOLINE_IMAGE"
+    value: "gcr.io/cloud-devrel-kokoro-resources/java8"
+}
+
+env_vars: {
+  key: "TRAMPOLINE_BUILD_FILE"
```
Comment (elharo) [inline .kokoro/continuous/dependencies.cfg]: we usually use two space indents, and this file is inconsistent

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -1,43 +1,93 @@
...
+to engage in behavior that violates the project’s code of conduct.
+
+If you see someone violating the code of conduct, you are encouraged to address
+the behavior directly with those involved. Many issues can be resolved quickly
+and easily, and this gives people more control over the outcome of their
+dispute. If you are unable to resolve the matter for any reason, or if the
+behavior is threatening or harassing, report it. We are dedicated to providing
+an environment where participants feel welcome and safe.
+
+Reports should be directed to *[PROJECT STEWARD NAME(s) AND EMAIL(s)]*, the
```
Comment (elharo) [inline CODE_OF_CONDUCT.md]: looks like these values should be replaced.

File: .kokoro/continuous/dependencies.cfg (hunk 1 above)
Comment (chingor13) [reply .kokoro/continuous/dependencies.cfg]: Whitespace in the templates are being fixed upstream: https://github.com/googleapis/synthtool/pull/275

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (raquellopez2112) [reply CODE_OF_CONDUCT.md]: T

Comment (chingor13) [reply CODE_OF_CONDUCT.md]: done

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+Thanks for stopping by to let us know something could be better!
+
+**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.
+
+Please run down the following list and make sure you've tried the usual "quick fixes":
+
+  - Search the issues already opened: https://github.com/googleapis/google-api-client/issues
+  - Check for answers on StackOverflow: http://stackoverflow.com/questions/tagged/google-cloud-platform
+
+If you are still having issues, please be sure to include as much information as possible:
```
Comment (elharo) [inline .github/ISSUE_TEMPLATE/bug_report.md]: delete "be sure to"

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+#### Environment details
+
+1. Specify the API at the beginning of the title (for example, "BigQuery: ...")
```
Comment (elharo) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Specify the API at the beginning of the title. For example, "BigQuery: ...".
General, Core, and Other are also allowed.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+```java
+// example
+```
+
+#### Stack trace
+```
+Any relevant stacktrace here.
+```
+
+#### External references such as API reference guides used
```
Comment (elharo) [inline .github/ISSUE_TEMPLATE/bug_report.md]: delete "used"

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+about: Suggest an idea for this library
+
+---
+
+Thanks for stopping by to let us know something could be better!
+
+**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.
+
+ **Is your feature request related to a problem? Please describe.**
+A clear and concise description of what the problem is. Example: I'm always frustrated when [...]
```
Comment (elharo) [inline .github/ISSUE_TEMPLATE/feature_request.md]: consider deleting "A clear and concise description of " here and below. The extra verbiage makes these instructions less clear and concise than they could be. :-)

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+ **Describe the solution you'd like**
+A clear and concise description of what you want to happen.
+ **Describe alternatives you've considered**
+A clear and concise description of any alternative solutions or features you've considered.
+ **Additional context**
+Add any other context or screenshots about the feature request here.
```
Comment (elharo) [inline .github/ISSUE_TEMPLATE/feature_request.md]: Add any --> Any
for parallelism

File: CODE_OF_CONDUCT.md [hunk 8]
```diff
@@ -1,43 +1,93 @@
...
-* Public or private harassment
-* Publishing other's private information,
-such as physical or electronic
-addresses, without explicit permission
-* Other unethical or unprofessional conduct.
+*   The use of sexualized language or imagery and unwelcome sexual attention or
+    advances
+*   Trolling, insulting/derogatory comments, and personal or political attacks
+*   Public or private harassment
+*   Publishing others' private information, such as a physical or electronic
```
Comment (elharo) [inline CODE_OF_CONDUCT.md]: "electronic address" is perhaps overreaching. I think it's perfectly OK to note someone's Twitter handle, blog URL, or developer email, for example.

Comment (chingor13) [reply CODE_OF_CONDUCT.md]: This CODE_OF_CONDUCT is straight from OSPO.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (chingor13) [reply .github/ISSUE_TEMPLATE/bug_report.md]: done

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
Comment (chingor13) [reply .github/ISSUE_TEMPLATE/bug_report.md]: done

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 4 above)
Comment (chingor13) [reply .github/ISSUE_TEMPLATE/bug_report.md]: done

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 6 above)
Comment (chingor13) [reply .github/ISSUE_TEMPLATE/feature_request.md]: done

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 7 above)
Comment (chingor13) [reply .github/ISSUE_TEMPLATE/feature_request.md]: done

File: README.md [hunk 9]
```diff
@@ -1,383 +1,29 @@
 # Google APIs Client Library for Java
 
-- [Library Maintenance](#maintenance)
```
Comment (kolea2) [inline README.md]: Is this getting moved into other modules?

Comment (chingor13) [reply README.md]: This content all moved into the wiki. I migrated all the content from developers.google.com and it was too much for the README.

**Pull request merged**